src/main/java/org/jitsi/jigasi/transcription/Participant.java

```diff
@@ -660,9 +660,15 @@ private void sendRequest(byte[] audio)
             {
                 session.sendRequest(request);
             }
+            else if (transcriber.getTranscriptionService().supportsStreamRecognition())
+            // re-establish prematurely ended streaming session
+            {
+                session = transcriber.getTranscriptionService()
+                        .initStreamingSession(this);
+                session.addTranscriptionListener(this);
+            }
```

Comment:
Shouldn't we here send the current request so we do not lose it?

Reply:
We have tried that, however, if the StreamingSession is not re-established immediately (host is restarting for instance), `sendRequest` will error out (`NullPointerException` )  and it will do that until the connexion is restablished. From our tests, this introduces undefined behaviour.

Reply:
But the code you have added initializes the session again(re-establish), and once that is done calling sendRequest will not work. Is this what you mean?

Reply:
Exactly.
Because if the initialization fails, sendRequest will be called and the transcription service will not be listening, resulting in a `NullPointerException`

Reply:
The thing is, without this fix, the connection can just randomly break and it will try to use the `sendSingleRequest` mode which is not supported by Google or VOSK, it just produces gibberish results.

Reply:
But what about when participant left and the session is ended? 
https://github.com/jitsi/jigasi/blob/78258390f4f35cbb562d3cd320b6bcf1188e412c/src/main/java/org/jitsi/jigasi/transcription/Participant.java#L539
This will re-establish the session ... which doesn't sound good